mod/groups/classes/Elgg/Groups/Routes/Profile.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Elgg\Groups\Routes;
+
+class Profile extends Index {
+
+	public function __construct($identifier, $segments) {
+		$this->identifier = $identifier;
+		$this->segments = $segments;
+	}
+
+	public function validate() {
+		$guid = $this->segments['guid'];
+		if (!elgg_entity_gatekeeper($guid, null, null, false)) {
+			return elgg_error_response('Group not found', 'groups', 404);
+		}
+
+		if (!elgg_group_gatekeeper(false, $guid)) {
+			return elgg_error_response('Group not accessible', 'groups', 403);
+		}
+	}
+
+	public function getBreadcrumbs() {
+		$breadcrumbs = parent::getBreadcrumbs();
+		$breadcrumbs[] = [
+			'title' => $this->getPageEntity()->getDisplayName(),
+			'href' => $this->getPageEntity()->getURL(),
+		];
+		return $breadcrumbs;
+	}
+
+	public function getContent() {
+		elgg_push_context('group_profile');
+
+		elgg_register_rss_link();
+
+		groups_register_profile_buttons($this->getPageEntity());
+
+		return elgg_view('groups/profile/layout', array('entity' => $this->getPageEntity()));
```

Comment:
Ideally, we would do something like:
```php
return new \ElggList::factory([
   'guids' => $this->getPageEntity()->guid,
   'item_view' => 'groups/profile/layout',
]);
```

JSON viewtype can then just do `echo json_encode($entity->toObject($some_context))`
Same for RSS.